cloud/pkg/cloudhub/servers/httpserver/certutil.go

```diff
@@ -126,3 +125,23 @@ func NewCertFromCa(cfg *certutil.Config, caCert *x509.Certificate, serverKey cry
 	}
 	return certDERBytes, nil
 }
+
+// ParseX509PrivateKey parse the private key from der format, support EC/PKCS1/PKCS8
+func ParseX509PrivateKey(der []byte) (crypto.Signer, error) {
+	caKeyEc, ecErr := x509.ParseECPrivateKey(der)
+	if ecErr == nil {
+		return caKeyEc, nil
+	}
+
+	caKeyPKCS1, pkcs1Err := x509.ParsePKCS1PrivateKey(der)
+	if pkcs1Err == nil {
+		return caKeyPKCS1, nil
+	}
```

Comment:
Is there a way to distinguish the encryption type of the key that needs to be parsed? Or to put it another way, if the PKCS1 secret key calls the ParseECPrivateKey method, an error will be reported, right?

Reply:
yeah
If the private key is not in the correct format, it will return err.
Parsing the private key format is the same as parsing the private key.

Reply:
We can distinguish the encryption type of the key from caCert.PublicKeyAlgorithm.

Reply:
[tls.X509KeyPair](https://github.com/golang/go/blob/go1.22.3/src/crypto/tls/tls.go#L245) is better. but look at [here](https://github.com/kubeedge/kubeedge/blob/v1.17.0/cloud/pkg/cloudhub/config/config.go#L44)